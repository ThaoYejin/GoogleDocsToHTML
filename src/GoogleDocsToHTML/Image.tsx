import { docs_v1 } from "googleapis";
import React from "react";

export interface ImageProps {
    element: docs_v1.Schema$ParagraphElement | undefined;
    doc: docs_v1.Schema$Document;
}

export const Image: React.FC<ImageProps> = ({
    element,
    doc
}) => {
    var inlineObjID: (string | undefined);

    //Get the ID of the image
    inlineObjID = element?.inlineObjectElement?.inlineObjectId;

    var inlineObj = doc?.inlineObjects;
    var source: (string | undefined);

    //Get the URL of the image
    if (inlineObjID != undefined && inlineObj != undefined) {
        source = inlineObj[inlineObjID].inlineObjectProperties?.embeddedObject?.imageProperties?.contentUri;
        return (
            <img src={source} />
        )
    }

    return (
        <></>
    )
}