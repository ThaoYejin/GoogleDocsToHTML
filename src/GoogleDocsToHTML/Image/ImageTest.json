{"title":"testGoogleDoc","body":{"content":[{"endIndex":1,"sectionBreak":{"sectionStyle":{"columnSeparatorStyle":"NONE","contentDirection":"LEFT_TO_RIGHT","sectionType":"CONTINUOUS"}}},{"startIndex":1,"endIndex":3,"paragraph":{"elements":[{"startIndex":1,"endIndex":2,"inlineObjectElement":{"inlineObjectId":"kix.5mgyfthwug6j","textStyle":{"weightedFontFamily":{"fontFamily":"Comic Sans MS","weight":400}}}},{"startIndex":2,"endIndex":3,"textRun":{"content":"\n","textStyle":{"weightedFontFamily":{"fontFamily":"Comic Sans MS","weight":400}}}}],"paragraphStyle":{"namedStyleType":"NORMAL_TEXT","direction":"LEFT_TO_RIGHT","indentFirstLine":{"unit":"PT"},"indentStart":{"unit":"PT"}}}}]},"documentStyle":{"background":{"color":{}},"pageNumberStart":1,"marginTop":{"magnitude":72,"unit":"PT"},"marginBottom":{"magnitude":72,"unit":"PT"},"marginRight":{"magnitude":72,"unit":"PT"},"marginLeft":{"magnitude":72,"unit":"PT"},"pageSize":{"height":{"magnitude":792,"unit":"PT"},"width":{"magnitude":612,"unit":"PT"}},"marginHeader":{"magnitude":36,"unit":"PT"},"marginFooter":{"magnitude":36,"unit":"PT"},"useCustomHeaderFooterMargins":true},"namedStyles":{"styles":[{"namedStyleType":"NORMAL_TEXT","textStyle":{"bold":false,"italic":false,"underline":false,"strikethrough":false,"smallCaps":false,"backgroundColor":{},"foregroundColor":{"color":{"rgbColor":{}}},"fontSize":{"magnitude":11,"unit":"PT"},"weightedFontFamily":{"fontFamily":"Arial","weight":400},"baselineOffset":"NONE"},"paragraphStyle":{"namedStyleType":"NORMAL_TEXT","alignment":"START","lineSpacing":115,"direction":"LEFT_TO_RIGHT","spacingMode":"COLLAPSE_LISTS","spaceAbove":{"unit":"PT"},"spaceBelow":{"unit":"PT"},"borderBetween":{"color":{},"width":{"unit":"PT"},"padding":{"unit":"PT"},"dashStyle":"SOLID"},"borderTop":{"color":{},"width":{"unit":"PT"},"padding":{"unit":"PT"},"dashStyle":"SOLID"},"borderBottom":{"color":{},"width":{"unit":"PT"},"padding":{"unit":"PT"},"dashStyle":"SOLID"},"borderLeft":{"color":{},"width":{"unit":"PT"},"padding":{"unit":"PT"},"dashStyle":"SOLID"},"borderRight":{"color":{},"width":{"unit":"PT"},"padding":{"unit":"PT"},"dashStyle":"SOLID"},"indentFirstLine":{"unit":"PT"},"indentStart":{"unit":"PT"},"indentEnd":{"unit":"PT"},"keepLinesTogether":false,"keepWithNext":false,"avoidWidowAndOrphan":true,"shading":{"backgroundColor":{}}}},{"namedStyleType":"HEADING_1","textStyle":{"fontSize":{"magnitude":20,"unit":"PT"}},"paragraphStyle":{"namedStyleType":"NORMAL_TEXT","direction":"LEFT_TO_RIGHT","spaceAbove":{"magnitude":20,"unit":"PT"},"spaceBelow":{"magnitude":6,"unit":"PT"},"keepLinesTogether":true,"keepWithNext":true}},{"namedStyleType":"HEADING_2","textStyle":{"bold":false,"fontSize":{"magnitude":16,"unit":"PT"}},"paragraphStyle":{"namedStyleType":"NORMAL_TEXT","direction":"LEFT_TO_RIGHT","spaceAbove":{"magnitude":18,"unit":"PT"},"spaceBelow":{"magnitude":6,"unit":"PT"},"keepLinesTogether":true,"keepWithNext":true}},{"namedStyleType":"HEADING_3","textStyle":{"bold":false,"foregroundColor":{"color":{"rgbColor":{"red":0.2627451,"green":0.2627451,"blue":0.2627451}}},"fontSize":{"magnitude":14,"unit":"PT"}},"paragraphStyle":{"namedStyleType":"NORMAL_TEXT","direction":"LEFT_TO_RIGHT","spaceAbove":{"magnitude":16,"unit":"PT"},"spaceBelow":{"magnitude":4,"unit":"PT"},"keepLinesTogether":true,"keepWithNext":true}},{"namedStyleType":"HEADING_4","textStyle":{"foregroundColor":{"color":{"rgbColor":{"red":0.4,"green":0.4,"blue":0.4}}},"fontSize":{"magnitude":12,"unit":"PT"}},"paragraphStyle":{"namedStyleType":"NORMAL_TEXT","direction":"LEFT_TO_RIGHT","spaceAbove":{"magnitude":14,"unit":"PT"},"spaceBelow":{"magnitude":4,"unit":"PT"},"keepLinesTogether":true,"keepWithNext":true}},{"namedStyleType":"HEADING_5","textStyle":{"foregroundColor":{"color":{"rgbColor":{"red":0.4,"green":0.4,"blue":0.4}}},"fontSize":{"magnitude":11,"unit":"PT"}},"paragraphStyle":{"namedStyleType":"NORMAL_TEXT","direction":"LEFT_TO_RIGHT","spaceAbove":{"magnitude":12,"unit":"PT"},"spaceBelow":{"magnitude":4,"unit":"PT"},"keepLinesTogether":true,"keepWithNext":true}},{"namedStyleType":"HEADING_6","textStyle":{"italic":true,"foregroundColor":{"color":{"rgbColor":{"red":0.4,"green":0.4,"blue":0.4}}},"fontSize":{"magnitude":11,"unit":"PT"}},"paragraphStyle":{"namedStyleType":"NORMAL_TEXT","direction":"LEFT_TO_RIGHT","spaceAbove":{"magnitude":12,"unit":"PT"},"spaceBelow":{"magnitude":4,"unit":"PT"},"keepLinesTogether":true,"keepWithNext":true}},{"namedStyleType":"TITLE","textStyle":{"fontSize":{"magnitude":26,"unit":"PT"}},"paragraphStyle":{"namedStyleType":"NORMAL_TEXT","direction":"LEFT_TO_RIGHT","spaceAbove":{"unit":"PT"},"spaceBelow":{"magnitude":3,"unit":"PT"},"keepLinesTogether":true,"keepWithNext":true}},{"namedStyleType":"SUBTITLE","textStyle":{"italic":false,"foregroundColor":{"color":{"rgbColor":{"red":0.4,"green":0.4,"blue":0.4}}},"fontSize":{"magnitude":15,"unit":"PT"},"weightedFontFamily":{"fontFamily":"Arial","weight":400}},"paragraphStyle":{"namedStyleType":"NORMAL_TEXT","direction":"LEFT_TO_RIGHT","spaceAbove":{"unit":"PT"},"spaceBelow":{"magnitude":16,"unit":"PT"},"keepLinesTogether":true,"keepWithNext":true}}]},"lists":{"kix.398gfgli44q7":{"listProperties":{"nestingLevels":[{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%0.","indentFirstLine":{"magnitude":18,"unit":"PT"},"indentStart":{"magnitude":36,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%1.","indentFirstLine":{"magnitude":54,"unit":"PT"},"indentStart":{"magnitude":72,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%2.","indentFirstLine":{"magnitude":90,"unit":"PT"},"indentStart":{"magnitude":108,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%3.","indentFirstLine":{"magnitude":126,"unit":"PT"},"indentStart":{"magnitude":144,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%4.","indentFirstLine":{"magnitude":162,"unit":"PT"},"indentStart":{"magnitude":180,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%5.","indentFirstLine":{"magnitude":198,"unit":"PT"},"indentStart":{"magnitude":216,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%6.","indentFirstLine":{"magnitude":234,"unit":"PT"},"indentStart":{"magnitude":252,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%7.","indentFirstLine":{"magnitude":270,"unit":"PT"},"indentStart":{"magnitude":288,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%8.","indentFirstLine":{"magnitude":306,"unit":"PT"},"indentStart":{"magnitude":324,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1}]}},"kix.6t36bbl3i3cb":{"listProperties":{"nestingLevels":[{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%0.","indentFirstLine":{"magnitude":18,"unit":"PT"},"indentStart":{"magnitude":36,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%1.","indentFirstLine":{"magnitude":54,"unit":"PT"},"indentStart":{"magnitude":72,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%2.","indentFirstLine":{"magnitude":90,"unit":"PT"},"indentStart":{"magnitude":108,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%3.","indentFirstLine":{"magnitude":126,"unit":"PT"},"indentStart":{"magnitude":144,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%4.","indentFirstLine":{"magnitude":162,"unit":"PT"},"indentStart":{"magnitude":180,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%5.","indentFirstLine":{"magnitude":198,"unit":"PT"},"indentStart":{"magnitude":216,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%6.","indentFirstLine":{"magnitude":234,"unit":"PT"},"indentStart":{"magnitude":252,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%7.","indentFirstLine":{"magnitude":270,"unit":"PT"},"indentStart":{"magnitude":288,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%8.","indentFirstLine":{"magnitude":306,"unit":"PT"},"indentStart":{"magnitude":324,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1}]}},"kix.9qdzu8oc8shs":{"listProperties":{"nestingLevels":[{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%0.","indentFirstLine":{"magnitude":18,"unit":"PT"},"indentStart":{"magnitude":36,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%1.","indentFirstLine":{"magnitude":54,"unit":"PT"},"indentStart":{"magnitude":72,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%2.","indentFirstLine":{"magnitude":90,"unit":"PT"},"indentStart":{"magnitude":108,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%3.","indentFirstLine":{"magnitude":126,"unit":"PT"},"indentStart":{"magnitude":144,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%4.","indentFirstLine":{"magnitude":162,"unit":"PT"},"indentStart":{"magnitude":180,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%5.","indentFirstLine":{"magnitude":198,"unit":"PT"},"indentStart":{"magnitude":216,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%6.","indentFirstLine":{"magnitude":234,"unit":"PT"},"indentStart":{"magnitude":252,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%7.","indentFirstLine":{"magnitude":270,"unit":"PT"},"indentStart":{"magnitude":288,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%8.","indentFirstLine":{"magnitude":306,"unit":"PT"},"indentStart":{"magnitude":324,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1}]}},"kix.cqlncuxpuhb9":{"listProperties":{"nestingLevels":[{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%0.","indentFirstLine":{"magnitude":18,"unit":"PT"},"indentStart":{"magnitude":36,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%1.","indentFirstLine":{"magnitude":54,"unit":"PT"},"indentStart":{"magnitude":72,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%2.","indentFirstLine":{"magnitude":90,"unit":"PT"},"indentStart":{"magnitude":108,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%3.","indentFirstLine":{"magnitude":126,"unit":"PT"},"indentStart":{"magnitude":144,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%4.","indentFirstLine":{"magnitude":162,"unit":"PT"},"indentStart":{"magnitude":180,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%5.","indentFirstLine":{"magnitude":198,"unit":"PT"},"indentStart":{"magnitude":216,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%6.","indentFirstLine":{"magnitude":234,"unit":"PT"},"indentStart":{"magnitude":252,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%7.","indentFirstLine":{"magnitude":270,"unit":"PT"},"indentStart":{"magnitude":288,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%8.","indentFirstLine":{"magnitude":306,"unit":"PT"},"indentStart":{"magnitude":324,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1}]}},"kix.d247awgrr7bj":{"listProperties":{"nestingLevels":[{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%0.","indentFirstLine":{"magnitude":18,"unit":"PT"},"indentStart":{"magnitude":36,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%1.","indentFirstLine":{"magnitude":54,"unit":"PT"},"indentStart":{"magnitude":72,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%2.","indentFirstLine":{"magnitude":90,"unit":"PT"},"indentStart":{"magnitude":108,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%3.","indentFirstLine":{"magnitude":126,"unit":"PT"},"indentStart":{"magnitude":144,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%4.","indentFirstLine":{"magnitude":162,"unit":"PT"},"indentStart":{"magnitude":180,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%5.","indentFirstLine":{"magnitude":198,"unit":"PT"},"indentStart":{"magnitude":216,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%6.","indentFirstLine":{"magnitude":234,"unit":"PT"},"indentStart":{"magnitude":252,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%7.","indentFirstLine":{"magnitude":270,"unit":"PT"},"indentStart":{"magnitude":288,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%8.","indentFirstLine":{"magnitude":306,"unit":"PT"},"indentStart":{"magnitude":324,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1}]}},"kix.fh0v6s6zo6br":{"listProperties":{"nestingLevels":[{"bulletAlignment":"START","glyphType":"GLYPH_TYPE_UNSPECIFIED","indentFirstLine":{"magnitude":18,"unit":"PT"},"indentStart":{"magnitude":36,"unit":"PT"},"textStyle":{"underline":false,"foregroundColor":{"color":{"rgbColor":{"red":0.2509804,"green":0.2509804,"blue":0.2509804}}},"fontSize":{"magnitude":15,"unit":"PT"}},"startNumber":1},{"bulletAlignment":"START","glyphType":"GLYPH_TYPE_UNSPECIFIED","indentFirstLine":{"magnitude":54,"unit":"PT"},"indentStart":{"magnitude":72,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"GLYPH_TYPE_UNSPECIFIED","indentFirstLine":{"magnitude":90,"unit":"PT"},"indentStart":{"magnitude":108,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"GLYPH_TYPE_UNSPECIFIED","indentFirstLine":{"magnitude":126,"unit":"PT"},"indentStart":{"magnitude":144,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"GLYPH_TYPE_UNSPECIFIED","indentFirstLine":{"magnitude":162,"unit":"PT"},"indentStart":{"magnitude":180,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"GLYPH_TYPE_UNSPECIFIED","indentFirstLine":{"magnitude":198,"unit":"PT"},"indentStart":{"magnitude":216,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"GLYPH_TYPE_UNSPECIFIED","indentFirstLine":{"magnitude":234,"unit":"PT"},"indentStart":{"magnitude":252,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"GLYPH_TYPE_UNSPECIFIED","indentFirstLine":{"magnitude":270,"unit":"PT"},"indentStart":{"magnitude":288,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"GLYPH_TYPE_UNSPECIFIED","indentFirstLine":{"magnitude":306,"unit":"PT"},"indentStart":{"magnitude":324,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1}]}},"kix.k2wetmbma9gu":{"listProperties":{"nestingLevels":[{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%0.","indentFirstLine":{"magnitude":18,"unit":"PT"},"indentStart":{"magnitude":36,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"○","glyphFormat":"%1","indentFirstLine":{"magnitude":54,"unit":"PT"},"indentStart":{"magnitude":72,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"■","glyphFormat":"%2","indentFirstLine":{"magnitude":90,"unit":"PT"},"indentStart":{"magnitude":108,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%3.","indentFirstLine":{"magnitude":126,"unit":"PT"},"indentStart":{"magnitude":144,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%4.","indentFirstLine":{"magnitude":162,"unit":"PT"},"indentStart":{"magnitude":180,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%5.","indentFirstLine":{"magnitude":198,"unit":"PT"},"indentStart":{"magnitude":216,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"DECIMAL","glyphFormat":"%6.","indentFirstLine":{"magnitude":234,"unit":"PT"},"indentStart":{"magnitude":252,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%7.","indentFirstLine":{"magnitude":270,"unit":"PT"},"indentStart":{"magnitude":288,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"END","glyphType":"ROMAN","glyphFormat":"%8.","indentFirstLine":{"magnitude":306,"unit":"PT"},"indentStart":{"magnitude":324,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1}]}},"kix.q6272utqy0uh":{"listProperties":{"nestingLevels":[{"bulletAlignment":"START","glyphSymbol":"●","glyphFormat":"%0","indentFirstLine":{"magnitude":90,"unit":"PT"},"indentStart":{"magnitude":108,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"○","glyphFormat":"%1","indentFirstLine":{"magnitude":126,"unit":"PT"},"indentStart":{"magnitude":144,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"■","glyphFormat":"%2","indentFirstLine":{"magnitude":162,"unit":"PT"},"indentStart":{"magnitude":180,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"●","glyphFormat":"%3","indentFirstLine":{"magnitude":198,"unit":"PT"},"indentStart":{"magnitude":216,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"○","glyphFormat":"%4","indentFirstLine":{"magnitude":234,"unit":"PT"},"indentStart":{"magnitude":252,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"■","glyphFormat":"%5","indentFirstLine":{"magnitude":270,"unit":"PT"},"indentStart":{"magnitude":288,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"●","glyphFormat":"%6","indentFirstLine":{"magnitude":306,"unit":"PT"},"indentStart":{"magnitude":324,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"○","glyphFormat":"%7","indentFirstLine":{"magnitude":342,"unit":"PT"},"indentStart":{"magnitude":360,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"■","glyphFormat":"%8","indentFirstLine":{"magnitude":378,"unit":"PT"},"indentStart":{"magnitude":396,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1}]}},"kix.vwbg2er8gd5w":{"listProperties":{"nestingLevels":[{"bulletAlignment":"START","glyphSymbol":"●","glyphFormat":"%0","indentFirstLine":{"magnitude":54,"unit":"PT"},"indentStart":{"magnitude":72,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphType":"ALPHA","glyphFormat":"%1.","indentFirstLine":{"magnitude":90,"unit":"PT"},"indentStart":{"magnitude":108,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"■","glyphFormat":"%2","indentFirstLine":{"magnitude":126,"unit":"PT"},"indentStart":{"magnitude":144,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"●","glyphFormat":"%3","indentFirstLine":{"magnitude":162,"unit":"PT"},"indentStart":{"magnitude":180,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"○","glyphFormat":"%4","indentFirstLine":{"magnitude":198,"unit":"PT"},"indentStart":{"magnitude":216,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"■","glyphFormat":"%5","indentFirstLine":{"magnitude":234,"unit":"PT"},"indentStart":{"magnitude":252,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"●","glyphFormat":"%6","indentFirstLine":{"magnitude":270,"unit":"PT"},"indentStart":{"magnitude":288,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"○","glyphFormat":"%7","indentFirstLine":{"magnitude":306,"unit":"PT"},"indentStart":{"magnitude":324,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1},{"bulletAlignment":"START","glyphSymbol":"■","glyphFormat":"%8","indentFirstLine":{"magnitude":342,"unit":"PT"},"indentStart":{"magnitude":360,"unit":"PT"},"textStyle":{"underline":false},"startNumber":1}]}}},"suggestionsViewMode":"PREVIEW_WITHOUT_SUGGESTIONS","inlineObjects":{"kix.5mgyfthwug6j":{"objectId":"kix.5mgyfthwug6j","inlineObjectProperties":{"embeddedObject":{"imageProperties":{"contentUri":"https://lh5.googleusercontent.com/dTGo1JBnJJXetdTHf5rxWkOgzhedEp1BwmQ8Ct7K73RjsauhFvVfPQgQ6k6drlGs5waTyr6_bkyz2QPhDQ4o02QhgZNPt7ZaRo3AGIR_vWfdp2RK5KLWtkUYG-GhJ7UeK0Oyc_unGCE1wB7f3g","cropProperties":{}},"embeddedObjectBorder":{"color":{"color":{"rgbColor":{}}},"width":{"unit":"PT"},"dashStyle":"SOLID","propertyState":"NOT_RENDERED"},"size":{"height":{"magnitude":234,"unit":"PT"},"width":{"magnitude":468,"unit":"PT"}},"marginTop":{"magnitude":9,"unit":"PT"},"marginBottom":{"magnitude":9,"unit":"PT"},"marginRight":{"magnitude":9,"unit":"PT"},"marginLeft":{"magnitude":9,"unit":"PT"}}}}},"documentId":"12rbiv-1ozz9maXRTYG-iYf41A5Goyul5TjyNIeHCpKI"}