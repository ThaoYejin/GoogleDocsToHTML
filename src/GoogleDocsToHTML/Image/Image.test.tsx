import React from 'react';
import renderer from 'react-test-renderer';
import {Image} from './Image';
import * as test from './ImageTest.json';

it('renders correctly', ()=>{
    var doc = test;
   var element ={
      "startIndex": 1,
      "endIndex": 2,
      "inlineObjectElement": {
        "inlineObjectId": "kix.5mgyfthwug6j",
        "textStyle": {
          "weightedFontFamily": {
            "fontFamily": "Comic Sans MS",
            "weight": 400
          }
        }
      }
    };

    const tree = renderer
    .create(<Image doc={doc} element={element}></Image>)
    .toJSON();
    expect(tree).toMatchSnapshot();
})