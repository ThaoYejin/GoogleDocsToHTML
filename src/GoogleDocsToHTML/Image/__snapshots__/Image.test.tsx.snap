// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  src="https://lh5.googleusercontent.com/dTGo1JBnJJXetdTHf5rxWkOgzhedEp1BwmQ8Ct7K73RjsauhFvVfPQgQ6k6drlGs5waTyr6_bkyz2QPhDQ4o02QhgZNPt7ZaRo3AGIR_vWfdp2RK5KLWtkUYG-GhJ7UeK0Oyc_unGCE1wB7f3g"
/>
`;
